specification/redisenterprise/resource-manager/Microsoft.Cache/preview/2020-10-01-preview/redisenterprise.json

```diff
@@ -0,0 +1,1652 @@
+          "enum": [
+            "Encrypted",
+            "Plaintext"
+          ],
+          "x-ms-enum": {
+            "name": "Protocol",
+            "modelAsString": true
+          }
+        },
+        "port": {
+          "type": "integer",
+          "format": "int32",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "description": "TCP port of the database endpoint. Specified at create time. Defaults to an available port."
+        },
+        "provisioningState": {
+          "$ref": "#/definitions/ProvisioningState",
+          "readOnly": true,
+          "description": "Current status of the database"
+        },
+        "clusteringPolicy": {
+          "type": "string",
+          "description": "Clustering policy - default is OSSCluster. Specified at create time.",
+          "enum": [
+            "EnterpriseCluster",
+            "OSSCluster"
+          ],
+          "x-ms-enum": {
+            "name": "ClusteringPolicy",
+            "modelAsString": true
+          },
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        },
+        "evictionPolicy": {
+          "type": "string",
+          "description": "Redis eviction policy - default is VolatileLRU",
+          "enum": [
+            "AllKeysLFU",
+            "AllKeysLRU",
+            "AllKeysRandom",
+            "VolatileLRU",
+            "VolatileLFU",
+            "VolatileTTL",
+            "VolatileRandom",
+            "NoEviction"
+          ],
+          "x-ms-enum": {
+            "name": "EvictionPolicy",
+            "modelAsString": true
+          }
+        },
+        "modules": {
+          "type": "array",
+          "description": "Optional set of redis modules to enable in this database - modules can only be added at creation time.",
+          "items": {
+            "$ref": "#/definitions/Module"
+          },
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        }
+      }
+    },
+    "DatabaseList": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Database"
+          },
+          "description": "List of databases"
+        },
+        "nextLink": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The URI to fetch the next page of results."
+        }
+      },
+      "description": "The response of a list-all operation."
+    },
+    "Module": {
+      "title": "Module settings",
+      "description": "Specifies configuration of a redis module",
+      "properties": {
+        "name": {
+          "type": "string",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "description": "The name of the module, e.g. 'RedisBloom', 'RediSearch', 'RedisTimeSeries'"
+        },
+        "args": {
```

Comment:
>args [](start = 9, length = 4)

This seems to specify a set of key value pairs as arguments to the module. If the user is deploying these via templates and wants to set a single value from a template parameter, it will be difficult for them to construct the string. Is it possible to model this as a key value object or an array of key/value tuples?

Reply:
It is based on the module configuration strings documented for the modules. For preview we don't have many parameters enabled and I don't see it as likely people will need to construct strings to set them based on their deployment template.

More background info:
https://oss.redislabs.com/redisbloom/Configuration/#passing-configuration-options-during-loading
https://oss.redislabs.com/redisearch/Configuring.html#passing_configuration_options_during_loading
https://oss.redislabs.com/redistimeseries/configuration/#passing-configuration-options-during-loading

Reply:
Sounds good.